src/table.ts

```diff
@@ -39,6 +38,9 @@ export class MemoryTable<T = any> extends DataSourceBase<T> implements IMemoryTa
         }
         return this._reg;
     }
+    get columns() {
+      return this.columnDefs.map(c => c.expression);
```

Comment:
the `columns` property is called A LOT.

I'm not really sure if that's premature optimization, but I chose to maintain a copy, rather than systematically copying a whole array each time `.columns` is accessed.

Reply:
Maybe worth then to make an third entity which would be responsible of maintaining the columns, which would be responsible for managing a cache for `columnDefs` and `columns`?

Not tested, but I think about something like this:
```js
class ColumnManager {
  private columnsByName = new Map();
  private _columnDefs: ColRef[]?;
  private _columns: IValue[]?;  
  
  get columns() {
    if (!this._columns) {
      this._columns = this.columnDefs.map(c => c.expression);
    }
    return this._columns;
  }

  get columnDefs() {
    if (!this._columnDefs) {
      this._columnDefs = [...this.columnsByName.values()];
    }
    return this._columnDefs;
  }

  invalidateCaches() {
    this._columns = this._columnDefs = null;
  }

  set(name, colDef) {
    if (this.has(name)) {
      this.invalidateCaches();
    } else {
      // Maybe this optimization is not worthy and we may invalidate the cache in any cases
      this._columns.push(colDef.expression);
      this._columnDefs.push(colDef);
    }
    return this.columnsByName.set(name, value);
  }

  has(name) {
    return this.columnsByName.has(name);
  }

  delete(name) {
    this.invalidateCache();
    return this.columnsByName.delete(name);
  }
}
```

Reply:
Yes, that would be cleaner :)

Reply:
Do you want a PR for this? (I think I can propose that during the evening)

Reply:
If you have some time, that would be awesome :)

Otherwise, I like the idea, so I will do it

Reply:
Done #100 :)